components/hooks/useDisplayMananger.ts

```diff
@@ -158,37 +158,60 @@ const useThreadViewDisplay = () => {
       }
     );
 
-    if (excludedNotices != null || activeFilters != null) {
+    if (filteredNotices.length > 0 || activeCategories.length > 0) {
       const finalDisplayPosts: PostType[] = [];
 
       displayPostsForView.forEach((post) => {
         if (
           post.tags.contentWarnings.some((tag) =>
-            excludedNotices?.includes(tag)
+            filteredNotices?.includes(tag)
           )
         ) {
           return;
         }
-        if (activeFilters == null) {
+        if (activeCategories.length == 0) {
           finalDisplayPosts.push(post);
         } else if (
-          activeFilters.includes(UNCATEGORIZED_LABEL) &&
+          activeCategories.includes(UNCATEGORIZED_LABEL) &&
           post.tags.categoryTags.length === 0
         ) {
           finalDisplayPosts.push(post);
         } else if (
-          post.tags.categoryTags.some((tag) => !!activeFilters.includes(tag))
+          post.tags.categoryTags.some((tag) => !!activeCategories.includes(tag))
         ) {
           finalDisplayPosts.push(post);
         }
       });
 
-      if (currentThreadViewMode !== THREAD_VIEW_MODE.THREAD) {
+      if (currentThreadViewMode == THREAD_VIEW_MODE.MASONRY) {
         return finalDisplayPosts;
       }
-      // Add all parents of posts, even if they don't have categories.
-      const displayPostsWithParents = [...finalDisplayPosts];
+      // Remove any posts whose ancestors include filtered cn's
+      const filteredFinalDisplayPosts: PostType[] = [];
       finalDisplayPosts.forEach((post) => {
+        let parent = post.parentPostId;
```

Comment:
I think the `break` is better, but the destructuring of the parent data is worse for readability because too many things are being called `post` in close proximity. So unless you feel strongly, I think this is the way to go: 
``` typescript
        let parentId = post.parentPostId;
        while (parentId != null) {
          const parentData = postsInfoMap.get(parentId)!;
          if (
            parentData.post.tags.contentWarnings.some((tag) =>
              filteredNotices?.includes(tag)
            )
          ) {
            // Error found! Stop the loop early.
            break;
          }
          parentId = parentData.parent?.postId || null;
        }
        if (parentId == null) {
          // We've traversed the whole parent chain without error, so
          // we can add the post to the final display.
          filteredFinalDisplayPosts.push(post);
        }
```

Reply:
LGTM!